state.go

```diff
@@ -643,22 +666,24 @@ func (usr *UpdateStatusReportState) Handle(ctx *StateContext, c Controller) (Sta
 		} else {
 			if usr.status == statusFailure {
 				log.Debugf("update failed, attempt log upload")
-				// TODO upload logs from the failed update, see
-				// https://tracker.mender.io/browse/MEN-437 for details
-				c.UploadLog(usr.update, []LogEntry{
-					LogEntry{
-						Timestamp: time.Now().Format(time.RFC3339),
-						Level:     "error",
-						Message:   "update failed",
-					},
-				})
+
+				logs, err := DeploymentLogger.GetLogs(usr.update.ID)
+				if err != nil {
+					log.Errorf("Failed to get deployment logs for deployment [%v]: %v",
+						usr.update.ID, err)
+				} else {
+					c.UploadLog(usr.update, logs)
```

Comment:
Actually, maybe you have some idea how to handle this error? We are already in the error state and if we can not upload logs there is anyway nothing we can do.